src/app/ReadHandler.h

```diff
@@ -572,6 +575,8 @@ class ReadHandler : public Messaging::ExchangeDelegate
 
     // TODO (#27675): Merge all observers into one and that one will dispatch the callbacks to the right place.
     Observer * mObserver = nullptr;
+
+    bool mUseLargePayloadBuffer = false;
```

Comment:
1. Is this the right place to put this boolean, in terms of packing?
2. Are we ever going to be in a situation where we need to allocate the buffer but don't have an exchange or a session to query this state from?  In other words, do we need a member at all?

Note the RAM increase from this PR on constrained systems...  Ideally there would be no RAM increase.

Reply:
boolean removed. The sessionHandle stored as a member in ReadHandler is used to fetch the correct size in ReadHandler.